packages/qwik/src/core/use/use-store.public.ts

```diff
@@ -93,15 +100,29 @@ export const useStore = <STATE extends object>(
     return get;
   }
   const value = isFunction(initialState) ? invoke(undefined, initialState) : initialState;
-  if (opts?.reactive === false) {
-    set(value);
-    return value;
-  } else {
-    const containerState = iCtx.$renderCtx$.$static$.$containerState$;
-    const recursive = opts?.deep ?? true;
-    const flags = recursive ? QObjectRecursive : 0;
-    const newStore = getOrCreateProxy(value, containerState, flags);
-    set(newStore);
-    return newStore;
+
+  if (!opts?.reactive || !opts?.bind) {
+    return value as STATE;
   }
-};
+
+  const containerState = iCtx.$renderCtx$.$static$.$containerState$;
+  const flags = opts?.deep ? QObjectRecursive : 0;
+  const newStore = getOrCreateProxy(value, containerState, flags);
+
+  if (opts?.bind) {
+    const bindFunctionsRecursively = (obj: any) => {
+      for (const [key, value] of Object.entries(obj)) {
+        if (typeof value === 'function') {
+          obj[key] = value.bind(newStore);
+        } else if (typeof value === 'object' && value !== null) {
+          bindFunctionsRecursively(value);
+        }
+      }
+    };
+
+    bindFunctionsRecursively(newStore);
```

Comment:
By calling this here, you're triggering all the listeners. You should implement this binding in `getOrCreateProxy`